### PR TITLE
NOISSUE - Update SDK and Things Service

### PR DESCRIPTION
**Pull request by rodneyosodo**
### What does this do?
1. Makes update thing to update either name or metadata and not both
2. Corrects pagemetadata for things and channels to include disconnected and connected
3. Adds reader URL for reading messages

### Which issue(s) does this PR fix/relate to?
N/A

### List any changes that modify/break current functionality
Listed

### Have you included tests for your changes?
Yes

### Did you document any new/modified functionality?
No

### Notes
N/A

### CODE REVIEW

File: things/postgres/things.go [hunk 1]
```diff
@@ -73,7 +73,21 @@ func (tr thingRepository) Save(ctx context.Context, ths ...things.Thing) ([]thin
 }
 
 func (tr thingRepository) Update(ctx context.Context, t things.Thing) error {
-	q := `UPDATE things SET name = :name, metadata = :metadata WHERE id = :id;`
+	var query []string
+	var upq string
+	if t.Name != "" {
```
Comment (manuio) [inline things/postgres/things.go]: Can we update with empty name or empty metadata anymore?

Comment (rodneyosodo) [reply things/postgres/things.go]: No

Comment (manuio) [reply things/postgres/things.go]: This means that if the name or the metadata has been edited there is no way to revert it to an empty value. Is this what we want?

Comment (rodneyosodo) [reply things/postgres/things.go]: Let me revert

File: cli/channels.go [hunk 2]
```diff
@@ -156,7 +156,7 @@ var cmdChannels = []cobra.Command{
 			pm := mfxsdk.PageMetadata{
 				Offset:       uint64(Offset),
 				Limit:        uint64(Limit),
-				Disconnected: false,
+				Disconnected: true,
```
Comment (manuio) [inline cli/channels.go:159]: Isn't `List of Channels connected to Thing` what we are using here? Why are you inverting this value?

File: cli/things.go [hunk 3]
```diff
@@ -189,7 +189,7 @@ var cmdThings = []cobra.Command{
 			pm := mfxsdk.PageMetadata{
 				Offset:       uint64(Offset),
 				Limit:        uint64(Limit),
-				Disconnected: true,
+				Disconnected: false,
```
Comment (manuio) [inline cli/things.go:192]: Same here, why are you inverting the value if it's to list `non-connected` ?

**Review by manuio — CHANGES_REQUESTED** [review]

File: cli/channels.go (hunk 2 above)
Comment (rodneyosodo) [reply cli/channels.go:159]: Here in `List of Things connected to a Channel`
when disconnected is set to true it returns a list of things not connected to that channel and when disconnected is set to false it returns a list of things connected to the channel

File: cli/things.go (hunk 3 above)
Comment (rodneyosodo) [reply cli/things.go:192]: Here `disconnected` should be `false` meaning `connected` it `true`

File: cli/channels.go (hunk 2 above)
Comment (manuio) [reply cli/channels.go:159]: Right, my bad. I got confused :)

**Review by manuio — APPROVED** [review]
LGTM

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**